### DISCUSSION TITLE
Set service name and provider?

### DISCUSSION THREAD

**Discussion by Chris230291** [streamlink/streamlink#3860] [Q&A]
### Checklist

- [X] This is a feature request and not a different kind of issue
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink)
- [X] [I have checked the list of open and recently closed plugin requests](https://github.com/streamlink/streamlink/issues?q=is%3Aissue+label%3A%22feature+request%22)

### Description

Is it possible for us to set the provider and service name of an output stream please?
Similar to ffmpegs `-metadata service_provider=myprovider -metadata service_name=myservice`

Thanks,
Chris

**Comment by bastimeyer** [comment]
Streamlink doesn't always re-mux its output stream, so it's not possible to set custom metadata. Pipe into ffmpeg via `streamlink --stdout ... | ffmpeg -i pipe:0 ...` and set it yourself this way. There are more than enough examples on the issue tracker and discussions here.
